### DISCUSSION TITLE
Problem when I just run tests in mnist example

### DISCUSSION THREAD

**Discussion by bsar321** [BindsNET/bindsnet#561] [General]
Hello,
In the example "eth_mnist.py" when I run the tests after training network I get an accuracy of 0.75 (75%). So I don't always have to do the training, I save the weights, assignments and proportions variables at the end of the training. At a later moment, if I perform only the test (giving the previously mentioned variables to the network) I get an accuracy of only 0.62 (62%). What am I forgetting to save and what should I provide to the network to be able to replicate the results without having to run the training every time?
Thanks
Bruno

**Comment by Hananel-Hazan** [comment]
Its seems that you saving all that needed to recreate the experiment. Try to save the variable you mention to a file and load them again when needed. I suspect that the weight or the assignments and proportions variables  change somehow. Also, make sure to use `network.train(mode=False)` in the end of the training.

**Reply by bsar321** [reply]
Hi, thank you for your reply.
What I do in the eth_mnist.py file is:
- After line 280 torch.save(network.connections[("X", "Ae")].w,'weights_INPUTtoEXCITATORY.pt') , and the same for assignments and proportions variables.
- Finishing the training, and after these saved files I do
network.connections[("X","Ae")].w = torch.load(weights_INPUTtoEXCITATORY.pt') and similar for the other parameters.

What I see that might be missing is the layer bias, but how do I access the bias?
Thank you again

**Reply by bsar321** [reply]
Hi,
I solved question. The missing variable was "theta" (Adaptative thresholds). 
Thank you

**Reply by Hananel-Hazan** [reply]
wonderful

**Reply by machuanfeng** [reply]
> Comment options
Hello, may I ask how to save this parameter (theta) and where is it located
